test/controller_inheritance_test.rb

```diff
@@ -0,0 +1,54 @@
+require "test_helper"
+
+class PostsInheritanceTest < ActionDispatch::IntegrationTest
+  test "it works" do
+    post = Post.create!
+
+    get("/alt_posts")
+
+    json = JSON.parse(@response.body)
+    assert_equal 1, json.size
+    assert_equal(post.id, json.first["id"])
+  end
+
+  test "it inherits filter from parent controller" do
+    post = Post.create!
+    Post.create!
+
+    get("/alt_posts", params: { filters: { id: post.id } })
+
+    json = JSON.parse(@response.body)
+    assert_equal 1, json.size
+    assert_equal post.id, json.first["id"]
+  end
+
+  test "it inherits sort from parent controller" do
+    Post.create!(title: "z")
+    Post.create!(title: "a")
+
+    get("/alt_posts", params: { sort: "title" })
+
+    json = JSON.parse(@response.body, object_class: OpenStruct)
+    assert_equal ["a", "z"], json.map(&:title)
+  end
+
+  test "it overrides filter from parent controller" do
+    Post.create!(body: "hi")
+    Post.create!(body: "hello")
+    Post.create!(body: "hola")
+    get("/alt_posts", params: { filters: { french_bread: ["hi", "hello"] } })
+
+    json = JSON.parse(@response.body)
+    assert_equal 2, json.size
+  end
+
+  test "it overrides sort from parent controller" do
+    Post.create!(priority: 3)
+    Post.create!(priority: 1)
+    Post.create!(priority: 2)
+    get("/alt_posts", params: { sort: "foobar" })
```

Comment:
Same comment here as above. It's not totally clear what the expectation is.

Reply:
Updated, see above.